src/implementation.ts

```diff
@@ -396,18 +416,22 @@ function getNonBatchLoader(resourceConfig: NonBatchResourceConfig, resourcePath:
             ${getLoaderTypeKey(resourceConfig, resourcePath)},
             ${getLoaderTypeVal(resourceConfig, resourcePath)}
         >(${getLoaderComment(resourceConfig, resourcePath)} async (keys) => {
-            const response = await Promise.all(keys.map(key => {
+            const responses = await Promise.all(keys.map(async key => {
                 if (typeof ${resourceReference} !== 'function') {
                     return Promise.reject([
                         '${errorPrefix(resourcePath)} ${resourceReference} is not a function.',
                         'Did you pass in an instance of ${resourcePath.join('.')} to "getLoaders"?',
                     ].join(' '));
                 }
```

Comment:
Maybe this should this be a regular `throw`? It can be a little confusing to mix Promises and async/await in the same function.

Reply:
ya we can do that (should be functionally equivalent)